tests/repo_finder/test_provenance_extractor.py

```diff
@@ -0,0 +1,457 @@
+    with pytest.raises(ProvenanceExtractionException):
+        _perform_provenance_comparison(payload, "", "")
+
+    # Remove repository key.
+    _json_modify(payload, ["predicate", "buildDefinition", "externalParameters", "sourceToBuild", "repository"], None)
+    with pytest.raises(ProvenanceExtractionException):
+        _perform_provenance_comparison(payload, "", "")
+
+    # Add repository back.
+    _json_modify(
+        payload,
+        ["predicate", "buildDefinition", "externalParameters", "sourceToBuild", "repository"],
+        target_repository,
+    )
+    # Re-test provenance validity.
+    _perform_provenance_comparison(payload, target_repository, target_commit)
+
+    # Remove commit.
+    _json_modify(payload, ["predicate", "buildDefinition", "resolvedDependencies"], None)
+    with pytest.raises(ProvenanceExtractionException):
+        _perform_provenance_comparison(payload, "", "")
+
+
+def test_slsa_v1_gcb_2(slsa_v1_gcb_2_provenance: str, target_repository: str, target_commit: str) -> None:
+    """Test SLSA v1 provenance with build type gcb and configSource."""
+    payload = json.loads(slsa_v1_gcb_2_provenance)
+    assert isinstance(payload, dict)
+    _perform_provenance_comparison(payload, target_repository, target_commit)
+
+    # Set repository to an empty string.
+    _json_modify(payload, ["predicate", "buildDefinition", "externalParameters", "configSource", "repository"], "")
+    with pytest.raises(ProvenanceExtractionException):
+        _perform_provenance_comparison(payload, "", "")
+
+    # Remove repository key.
+    _json_modify(payload, ["predicate", "buildDefinition", "externalParameters", "configSource", "repository"], None)
+    with pytest.raises(ProvenanceExtractionException):
+        _perform_provenance_comparison(payload, "", "")
+
+    # Re-add repository key with a bad value.
+    _json_modify(payload, ["predicate", "buildDefinition", "externalParameters", "configSource", "repository"], "bad")
+    with pytest.raises(ProvenanceExtractionException):
+        _perform_provenance_comparison(payload, "", "")
+
+
+def test_slsa_v1_github(slsa_v1_github_provenance: str, target_repository: str, target_commit: str) -> None:
+    """Test SLSA v1 provenance with build type GitHub."""
+    payload = json.loads(slsa_v1_github_provenance)
+    assert isinstance(payload, dict)
+    _perform_provenance_comparison(payload, target_repository, target_commit)
+
+    # Set repository to an empty string.
+    _json_modify(payload, ["predicate", "buildDefinition", "externalParameters", "workflow", "repository"], "")
+    with pytest.raises(ProvenanceExtractionException):
+        _perform_provenance_comparison(payload, "", "")
+
+    # Remove repository key.
+    _json_modify(payload, ["predicate", "buildDefinition", "externalParameters", "workflow", "repository"], None)
+    with pytest.raises(ProvenanceExtractionException):
+        _perform_provenance_comparison(payload, "", "")
```

Comment:
Please use `@pytest.mark.parametrize` for tests like these where you want to test multiple different cases of input with the same test logic. It will make the test easier to manage since it separates test data from test logic, as well as allowing for the test logic to be defined once instead of repeatedly. I am also not a big fan of mutating test data like this to test different cases, and I prefer to create the input once each time with `@pytest.mark.parametrize`. For example, each case in this test can take a pair of `(keys, new_value)` as input, modify the payload accordingly only once, and do the assertion. I think that is enough for the logic of a unit test function like this. 

In addition, if the test logic deviates between cases, we should separate them into separate test functions. This test, for example, can also be split into two test functions: one to test for valid provenance cases, and the other for invalid cases.